File: dotnet/DesignPatterns/StructuralDesignPatterns/CompositeDesignPattern/CompositeDesignPattern.Example3/Employee.cs
namespace CompositeDesignPattern.Example3;

public abstract class Employee
{
	protected string _name;
	protected double _salary;
	protected Employee(string name, double salary)
	{
		_name = name;
		_salary = salary;
	}

	public string GetName() => _name;
	public double GetSalary() => _salary;
	public abstract void Print();
}

File: java/DesignPatterns/StructuralDesignPatterns/CompositeDesignPattern/CompositeDesignPattern.Example3/src/main/java/structural/composite/example3/Employee.java
package structural.composite.example3;

public abstract class Employee {
    protected String name;
    protected double salary;

    protected Employee(String name, double salary) {
        this.name = name;
        this.salary = salary;
    }

    public String getName() { return name; }
    public double getSalary() { return salary; }
    public abstract void print();
}
